Code/Engine/Core/World/Implementation/GameObject_inl.h

```diff
@@ -99,7 +99,7 @@ EZ_ALWAYS_INLINE void ezGameObject::SetName(const ezHashedString& sName)
 EZ_ALWAYS_INLINE void ezGameObject::SetGlobalKey(const char* szGlobalKey)
 {
   ezHashedString sGlobalKey;
-  sGlobalKey.Assign(szGlobalKey);
+  sGlobalKey.Assign(ezStringView(szGlobalKey));
```

Comment:
Currently you are adding lots of unnecessary casts. Once you touch Core, szGlobalKey would also change to an ezStringView and then you just have noise left. I think you should only compile Foundation, fix things there, then disable the explicit constructor again and then make sure that the rest still compiles. And revert any such redundant casts.

Reply:
These casts are not unnecessary. They are necessary and they are exactly what the compiler would do implicitly without the explicit constructor. Just in this case you can clearly see what is going on (because the ezStringView constructor is called explicitly). I clearly prefer this version because you can see from it what is going on (length counting) and you can even grep for these occurances to find places for improvement.